lib/vrl/stdlib/src/parse_grok.rs

```diff
@@ -115,6 +134,47 @@ impl Function for ParseGrok {
             remove_empty,
         }))
     }
+
+    fn compile_argument(
+        &self,
+        _args: &[(&'static str, Option<FunctionArgument>)],
+        _ctx: &mut FunctionCompileContext,
+        name: &str,
+        expr: Option<&expression::Expr>,
+    ) -> CompiledArgument {
+        match (name, expr) {
+            ("pattern", Some(expr)) => {
+                let pattern = expr
+                    .as_value()
+                    .unwrap() // TODO is this unwrap really bad
```

Comment:
I need answers! 😂

Reply:
It is!

Reply:
Actually it isn't that bad, because currently we are still calling the  `compile` function which catches that the expression is not a static before we get to this point. But once that code is changed, thin this will cause a panic. So I'll fix it now.